test/common/filesystem/watcher_impl_test.cc

```diff
@@ -91,6 +91,25 @@ TEST_F(WatcherImplTest, Create) {
   dispatcher_->run(Event::Dispatcher::RunType::Block);
 }
 
+TEST_F(WatcherImplTest, Modify) {
+  Filesystem::WatcherPtr watcher = dispatcher_->createFilesystemWatcher();
+
+  TestUtility::createDirectory(TestEnvironment::temporaryPath("envoy_test"));
+  std::ofstream file(TestEnvironment::temporaryPath("envoy_test/watcher_target"));
+
+  WatchCallback callback;
+  EXPECT_CALL(callback, called(Watcher::Events::Modified));
```

Comment:
Move this EXPECT_CALL to right before where you append to the file?

Also I think you also want to declare testing::InSequence as the first line of this test.

Reply:
I followed the pattern used in the other tests in this suite, which all appear to declare their `EXPECT_CALL`s right after before `watcher->addWatch()` is called. Given that, would you like me to leave this as-is, change it for just this new test, or revise all of the existing tests?

As for `testing::InSequence`, I don't think that would make much of a difference here because this test only makes a single assertion/expectation. I'm happy to add it for correctness, though. Would you like me to add that to the existing tests that would benefit from it, too?

Reply:
I'm not sure about InSequence but I think if you move the EXPECT_CALL to right before line 109, then the test would fail if the Modified event were called in the first call to dispatcher_->run().

It would be nice to also make similar improvements in precision to the other tests but if that winds up taking you down a rathole I'd be happy with this one.

Reply:
Makes sense to me. 👍